app/services/induction/schedule_for_new_cohort.rb

```diff
@@ -1,22 +1,26 @@
 # frozen_string_literal: true
 
 class Induction::ScheduleForNewCohort < BaseService
+  EXTENDED_SCHEDULE_IDENTIFIER = "ecf-extended-september"
+
   def call
-    if induction_record && induction_record.cohort == cohort
-      induction_record.schedule
-    else
-      Finance::Schedule::ECF.find_by(cohort:,
-                                     schedule_identifier: induction_record&.schedule_identifier) ||
-        Finance::Schedule::ECF.default_for(cohort:)
-    end
+    return induction_record.schedule if induction_record && induction_record.cohort == cohort
+
+    Finance::Schedule::ECF.find_by(cohort:, schedule_identifier:) ||
+      Finance::Schedule::ECF.default_for(cohort:)
   end
 
 private
 
-  attr_reader :cohort, :induction_record
+  attr_reader :cohort, :induction_record, :cohort_changed_after_payments_frozen
 
-  def initialize(cohort:, induction_record:)
+  def initialize(cohort:, induction_record:, cohort_changed_after_payments_frozen: false)
     @cohort = cohort
     @induction_record = induction_record
+    @cohort_changed_after_payments_frozen = cohort_changed_after_payments_frozen
```

Comment:
Would it not be "safer" to look for the value of `cohort_changed_after_payments_frozen` via `induction_record.participant_profiile` so it cannot be forgotten when invoking the service or is the point that this is used that flag isn't yet set on the participant profile?

Reply:
Exactly, the flag has not been set yet in the participant profile of the induction record.